corehq/apps/export/export.py

```diff
@@ -136,22 +138,35 @@ def get_export_file(export_instances, filters, progress_tracker=None):
     with writer.open(export_instances):
         for export_instance in export_instances:
             # TODO: Don't get the docs multiple times if you don't have to
-            docs = _get_export_documents(export_instance, filters)
-            _write_export_instance(writer, export_instance, docs, progress_tracker)
+            for docs in _iter_export_documents(export_instance, filters):
+                _write_export_instance(writer, export_instance, docs, progress_tracker)
 
     return ExportFile(writer.path, writer.format)
 
 
-def _get_export_documents(export_instance, filters):
+def _iter_export_documents(export_instance, filters):
     query = _get_base_query(export_instance)
     for filter in filters:
         query = query.filter(filter.to_es_filter())
+
+    if export_instance.type == CASE_EXPORT:
+        user_id_term = 'user_id'
+    elif export_instance.type == FORM_EXPORT:
+        user_id_term = 'form.meta.userID'
+    else:
+        user_id_term = None
+
     # size here limits each scroll request, not the total number of results
-    return query.size(1000).scroll()
+    for chunked_query in chunk_query(query, user_id_term, chunk_size=USER_ID_CHUNK_SIZE):
+        yield chunked_query.size(1000).scroll()
```

Comment:
that's a good idea, i was hesitant at first since i thought there'd be a lot of magic crammed into a function (chunking a term and scrolling) and they seemed like different things. thinking about it more i'm warming up to it